internal/circuit/transport_test.go

```
@@ -0,0 +1,597 @@
+		WindowSeconds:       60,
+		CooldownSeconds:     300,
+		MaxTransientRetries: 1,
+		MaxRateLimitRetries: 1,
+	}.Defaults()
+	return NewTransport(inner, store, cfg, "openai", nil)
+}
+
+// dummyRequest creates a minimal POST request with a cacheable body.
+func dummyRequest() *http.Request {
+	body := `{"model":"gpt-4o","messages":[]}`
+	req := httptest.NewRequest(http.MethodPost, "/openai/v1/chat/completions",
```

golangci-lint (2.11.4)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)